autoload/lsp/internal/inlay_hints.vim

```diff
@@ -0,0 +1,95 @@
+let s:use_vim_textprops = lsp#utils#_has_textprops() && !has('nvim')
+
+function! s:set_inlay_hints(data) abort
+    let l:bufnr = bufnr('%')
+
+    call s:clear_inlay_hints()
+
+    if mode() !=# 'n' | return | endif
+
+    if lsp#client#is_error(a:data['response']) | return | endif
+
+    " Get hints from the response
+    let l:hints = a:data['response']['result']
+    if empty(l:hints)
+        return
+    endif
+
+    for l:hint in l:hints
+        let l:text = (get(l:hint, 'paddingLeft', v:false) ? ' ' : '') . l:hint.label[0].value . (get(l:hint, 'paddingRight', v:false) ? ' ' : '')
+        if l:hint.kind ==# 1
```

Comment:
should handle optional kind. Though not sure what a reasonable default here would be.

```
/**
	 * The kind of this hint. Can be omitted in which case the client
	 * should fall back to a reasonable default.
	 */
	kind?: InlayHintKind;
```